text/0018-publish-update.md

```diff
@@ -0,0 +1,180 @@
+
+## Update
+Update brings your data to the latest known version of a dataset. Update comes in two flavors:
+* Updating Someone else's dataset
+* Updating A Dataset You Own
+Both types of update will be accessed via the same command like and API interfaces, they only differ based on the dataset reference in question.
+
+#### Updating a peer's dataset
+Running update on a peer dataset checks to see if the peer is online and calls `get_dataset`, if the version is different, it'll grab that dataset. If the peer in question is offline we can't check for updates. Once publish pushes to the registry we can check the registry for updates.
+
+Later on we'll want to do proactive checks for updates by submitting a list of dataset refs we'd like to grab updates for to peers and registries.
+
+#### Updating a dataset you own
+running update on a dataset in your namespace will check if a transform script is specified, If so, it'll re-execute the transform & check for changes. If changes exist it'll create a commit.
+
+Manual changes to a dataset are still run with `qri save`. Calling update on a dataset with no transform script is an error. That error should tell users to run `qri save`.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+publish and update are both steps that require an existing dataset.
+
+## Publish:
+#### CLI:
+```
+$ qri publish --help
+
+Description:
+  Qri publish makes your dataset available to others. While online peers that 
+  connect to you can only see datasets and versions that you've published. 
+  publishing a dataset always makes all previous history entries available.
+
+  By default publish also publishes your dataset to all configured registries,
+  to publish to only specific registries, use the --
+
+Examples:
+  # publish the latest version of a dataset
+  $ qri publish me/dataset
+
+  # publish a dataset (and all previous versions) at a specific version
+  $ qri publish me/dataset@/ipfs/QmFoo...
+
+  # unpublish a dataset
+  $ qri publish -d me/dataset
+
+Flags:
+  --delete -d       unpublish the specified dataset
+  --no-registry -r  don't publish to the registry
+  --registries      comma separated list of registries to publish to, overrides --no-registry
+```
+
+#### API:
+```
+# publish a reference:
+POST http://localhost:2505/publish/me/dataset
+
+# publish a dataset (and all previous versions) at a specific version
+POST http://localhost:2505/publish/me/dataset/at/ipfs/QmFoo...
+
+# unpublish a dataset
+DELETE http://localhost:2505/publish/me/dataset
+```
+
+## Update:
+#### CLI:
+```
+$ qri update --help
+
+Description:
+  Update fast-forwards your dataset to the latest known version. If the dataset
+  is owned by a peer, update will ask the peer for any new versions and download
```

Comment:
"If the dataset is not in your namespace, update will ask the peer"

"If the dataset is in a peer's namespace, update will ask that peer"